### ISSUE TITLE
run_on_all_* functions don't work with schemas

### ISSUE THREAD

**Issue by samay-sharma** [citusdata/citus#1339]
Steps to reproduce:

```
create schema test_schema;
create table test_schema.test_table ( col1 int);
select run_command_on_shards('test_schema.test_table','select pg_table_size(''%I'')');
```

This displays the following:
```
                       run_command_on_shards                        
--------------------------------------------------------------------
 (102072,f,"ERROR:  relation ""test_table_102072"" does not exist")
 (102073,f,"ERROR:  relation ""test_table_102073"" does not exist")
 (102074,f,"ERROR:  relation ""test_table_102074"" does not exist")
 (102075,f,"ERROR:  relation ""test_table_102075"" does not exist")
 (102076,f,"ERROR:  relation ""test_table_102076"" does not exist")
 (102077,f,"ERROR:  relation ""test_table_102077"" does not exist")
 (102078,f,"ERROR:  relation ""test_table_102078"" does not exist")
 (102079,f,"ERROR:  relation ""test_table_102079"" does not exist")
 (102080,f,"ERROR:  relation ""test_table_102080"" does not exist")
 (102081,f,"ERROR:  relation ""test_table_102081"" does not exist")
 (102082,f,"ERROR:  relation ""test_table_102082"" does not exist")
 (102083,f,"ERROR:  relation ""test_table_102083"" does not exist")
 (102084,f,"ERROR:  relation ""test_table_102084"" does not exist")
 (102085,f,"ERROR:  relation ""test_table_102085"" does not exist")
 (102086,f,"ERROR:  relation ""test_table_102086"" does not exist")
 (102087,f,"ERROR:  relation ""test_table_102087"" does not exist")
 (102088,f,"ERROR:  relation ""test_table_102088"" does not exist")
 (102089,f,"ERROR:  relation ""test_table_102089"" does not exist")
 (102090,f,"ERROR:  relation ""test_table_102090"" does not exist")
 (102091,f,"ERROR:  relation ""test_table_102091"" does not exist")
 (102092,f,"ERROR:  relation ""test_table_102092"" does not exist")
 (102093,f,"ERROR:  relation ""test_table_102093"" does not exist")
 (102094,f,"ERROR:  relation ""test_table_102094"" does not exist")
 (102095,f,"ERROR:  relation ""test_table_102095"" does not exist")
 (102096,f,"ERROR:  relation ""test_table_102096"" does not exist")
 (102097,f,"ERROR:  relation ""test_table_102097"" does not exist")
 (102098,f,"ERROR:  relation ""test_table_102098"" does not exist")
 (102099,f,"ERROR:  relation ""test_table_102099"" does not exist")
 (102100,f,"ERROR:  relation ""test_table_102100"" does not exist")
 (102101,f,"ERROR:  relation ""test_table_102101"" does not exist")
 (102102,f,"ERROR:  relation ""test_table_102102"" does not exist")
 (102103,f,"ERROR:  relation ""test_table_102103"" does not exist")
(32 rows)
```

Worker node logs:
```
2017-04-18 23:37:34.007 UTC [21542][6/14] : [1-1] [app=citus] ERROR:  relation "test_table_102083" does not exist at character 22
2017-04-18 23:37:34.007 UTC [21542][6/14] : [2-1] [app=citus] STATEMENT:  select pg_table_size('test_table_102083')
```

Manually including the schema name in the string works as a workaround, but this should work by itself.

```
select run_command_on_shards('test_schema.test_table','select pg_table_size(''test_schema.%I'')');
```

**Comment by marcocitus** [comment]
The underlying problem is that `shard_name` doesn't add a schema name. #1266 introduces a `qualified_shard_name` function with which we can resolve this by redefining the `run_on_all_*` functions to use `qualified_shard_name` if the table is not in the public schema.

One question is whether this breaks any existing scripts and whether it's ok to break them, e.g. the above workaround would not work if we start adding schema names.

**Comment by samay-sharma** [comment]
Good point. I used it with schemas for the first time this week. I have recommended these functions a few times but don't recall using schema any time else.

But, you can't ever know what people run when they get access to these functions ;) (Speaking from personal experience :D)

**Comment by marcocitus** [comment]
Fixed by #1266
